src/typeahead/typeahead.js

```diff
@@ -435,6 +437,8 @@ angular.module('ui.bootstrap.typeahead', ['ui.bootstrap.position'])
 
         if (appendToBody) {
           $document.find('body').append($popup);
+        } else if (appendToElementId !== false) {
```

Comment:
Given the above setting of the variable to `false` by default, I think this is a fine check.